### ISSUE TITLE
Disqus thread IDs need to use GUIDs instead of INT

### ISSUE THREAD

**Issue by Shazwazza** [Shazwazza/Articulate#102]
INTs cause issues when migrating content, specifically if you are using Courier, etc.. that matches by GUID and INT ids cannot be guaranteed.

This also means we need to update the Disqus import feature, etc... 

I'm in the process of seeing if i can force my current Disqus comments to be updated with a new unique ID, but not sure if that's even possible, we'll see.

**Comment by Shazwazza** [comment]
Ok so what I've done in my current solution is change the disqus_identifier to be the GUID of the content item, this will of course work for all content regardless of which environment so long as the GUIDs remain the same between them (i.e. like Courier does)

I've deployed this to my own site and temporary site (which will become my new live site). The issue here is that Disqus needs to be updated with all of the new GUID Ids instead of INT Ids. The way that I did this (not sure if there's a better way but it seems to work):

* Update your code to use the GUIDs instead of IDs for the disqus_identifier (everywhere)
* On your disqus account go to the URL: /admin/discussions/migrate/ which will bring up the migrating tools
* Click Start Url Mapper, download the CSV 
* Update the CSV with the correct format (i.e. sourceURL, destinationURL) ... This is also a good chance to see if any of your posts are tagged with an incorrect fully qualified URL.
* Upload the CSV. This will tell Disqus to update all your links and in the meantime, I think update all IDs to GUIDs

Then I also clicked on the StartCrawler link and let that do it's thing.

After a few hours, it looks like all of my posts now have the correct comments and comment counts.

Next thing I need to do is fix this in Articulate Core. However, until Umbraco 7.3 is released we need to lookup the GUID for each INT. This is a single lookup at a time, however it is cached... but this does mean that on the archive pages that show all of the blog posts, it will make quite a lot of SQL calls one time. I'll investigate a better way to achieve this hopefully.
